src/functions.php

```diff
@@ -27,24 +26,10 @@ function filesystem(?FilesystemDriver $driver = null): Filesystem
             return $map[$loop];
         }
 
-        $defaultDriver = createDefaultDriver();
-
-        if (!\defined("AMP_WORKER")) { // Prevent caching in workers, cache in parent instead.
-            $defaultDriver = new StatusCachingFilesystemDriver($defaultDriver);
-        }
-
-        $filesystem = new Filesystem($defaultDriver);
-    } else {
-        $filesystem = new Filesystem($driver);
-    }
-
-    if (\defined("AMP_WORKER") && $driver instanceof ParallelFilesystemDriver) {
-        throw new \Error("Cannot use the parallel driver within a worker");
+        $driver = new StatusCachingFilesystemDriver(createDefaultDriver());
```

Comment:
I'm not sure what to do about this. The caching in multiple workers isn't really safe.

Reply:
Good spot. You're right, we should only cache in the parent to avoid stale or conflicting caches in workers.